src/react-components/code-sample.tsx

```diff
@@ -11,16 +11,24 @@ import classes from "./code-sample.styles.js";
 import { CodeViewer, ErrorDisplay } from "./code-viewer.js";
 import codehikeTheme from "./codehike/theme.js";
 import { CopyButton } from "./copy-button.js";
-import { LanguageSelector } from "./language-selector.js";
 import { LanguageSelectorSkeleton, LoadingSkeleton } from "./skeleton.js";
 import { getCssVars, useSystemColorMode } from "./styles.js";
-import { type CodeSampleTitleComponent, CodeSampleTitle } from "./titles.js";
+import {
+  CodeSampleFilenameTitle,
+  CodeSampleTitle,
+  type CodeSampleTitleComponent,
+} from "./titles.js";
+import { prettyLanguageName } from "./utils.js";
+import { Selector } from "./selector";
 
 export type CodeSamplesViewerProps = {
   /** Whether the code snippet should be copyable. */
   copyable?: boolean;
-  /** Default language to show in the code playground. */
-  defaultLang?: string;
+
+  /** Default language to show in the code playground. If not found in the snippets, the first one will be used. */
+  defaultLanguage?: string;
+  /** Default operation to show in the code playground. If not found in the snippets, the first one will be used. */
+  defaultOperation?: string;
```

Comment:
should mirror the `operations` field's type def? Since a code sample may not have an operationId.